cobra/core/solution.py

```diff
@@ -292,11 +292,17 @@ def get_solution(model, reactions=None, metabolites=None):
     var_primals = model.solver.primal_values
     reduced = zeros(len(reactions))
     var_duals = model.solver.reduced_costs
-    for (i, rxn) in enumerate(reactions):
-        forward = rxn.forward_variable.name
-        reverse = rxn.reverse_variable.name
-        fluxes[i] = var_primals[forward] - var_primals[reverse]
-        reduced[i] = var_duals[forward] - var_duals[reverse]
+    # reduced costs are not always defined, e.g. for integer problems
+    if var_duals[0] is None:
```

Comment:
I don't think that will work since you need to index with a reaction id. There is `six.next(six.itervalues(var_duals))` but it's a bit ugly :thinking: